packages/lib/gpcircuits/test/bounds.spec.ts

```diff
@@ -0,0 +1,268 @@
+import { WitnessTester } from "circomkit";
+import "mocha";
+import {
+  BoundsCheckModuleInputNamesType,
+  BoundsCheckModuleOutputNamesType
+} from "../src";
+import { circomkit } from "./common";
+
+function inInterval(input: bigint, bounds: bigint[]): boolean {
+  return bounds[0] <= input && input <= bounds[1];
+}
+
+// 'Critical intervals' are those intervals of field elements which are larger
+// than `nBits` bits yet still satisfy the constraints of the `InInterval`
+// circuit for given bounds [a,b].  These are the intervals [1 << nBits, 1 <<
+// nBits + a - 1] and [p - (1 << nBits) + b + 1, p-1], where p is the field
+// modulus. This procedure returns these as an array of closed intervals, where
+// empty intervals are omitted.
+function criticalIntervals(
+  bounds: [bigint, bigint],
+  nBits: number
+): [bigint, bigint][] {
+  if (
+    bounds.some((bound) => !inInterval(bound, [0n, (1n << BigInt(nBits)) - 1n]))
+  ) {
+    throw new TypeError("Invalid bounds");
+  }
+  const [a, b] = bounds;
+  return (
+    [
+      [1n << BigInt(nBits), (1n << BigInt(nBits)) + a - 1n],
+      [
+        21888242871839275222246405745257275088548364400416034343698204186575808495617n -
```

Comment:
Are these big constants p and p-1?  Can you import them rather than using opaque ints?